src/pytorch_lightning/tuner/lr_finder.py

```diff
@@ -380,10 +392,20 @@ def on_train_batch_end(
         # Check if we diverging
         if self.early_stop_threshold is not None:
             if current_step > 1 and smoothed_loss > self.early_stop_threshold * self.best_loss:
-                trainer.fit_loop.max_steps = current_step  # stop signal
+                trainer.should_stop = True  # stop signal
                 if self.progress_bar:
                     self.progress_bar.close()
 
+        if trainer.should_stop:
```

Comment:
can you add a todo that this should be replaced with reduce_boolean_decision (any) after #15253 lands?